site/en/docs/workbox/using-plugins/index.md

```diff
@@ -0,0 +1,149 @@
+- [`requestWillFetch`](/docs/workbox/reference/workbox-core/#method-RequestWillFetchCallback): Called whenever a request is about to go to the network. Useful when you need to change the [`Request`](https://developer.mozilla.org/docs/Web/API/Request) just before it goes to the network.
+- [`fetchDidFail`](/docs/workbox/reference/workbox-core/#method-FetchDidFailCallback): Called when a network request fails, most likely due to an absence of network connectivity, and will not fire when the browser has a network connection, but receives an error (for example, `404 Not Found`).
+- [`fetchDidSucceed`](/docs/workbox/reference/workbox-core/#method-FetchDidSucceedCallback): Called whenever a network request succeeds, regardless of the HTTP response code.
+- [`handlerWillStart`](/docs/workbox/reference/workbox-core/#method-HandlerWillStartCallback): Called before any handler logic starts running, which is useful if you need to set the initial handler state. For example, if you wanted to know how long the handler took to generate a response, you could make a note of the start time in this callback.
+- [`handlerWillRespond`](/docs/workbox/reference/workbox-core/#method-HandlerWillRespondCallback): Called before the strategy's [`handle()` method](/docs/workbox/reference/workbox-strategies/#method-CacheFirst-handle) returns a response, which is helpful if you need to modify a response before returning it to a [`RouteHandler`](/docs/workbox/reference/workbox-core/#type-RouteHandler) or other custom logic.
+- [`handlerDidRespond`](/docs/workbox/reference/workbox-core/#method-HandlerDidRespondCallback): Called after the strategy's [`handle()` method](/docs/workbox/reference/workbox-strategies/#method-CacheFirst-handle) returns a response. This is when it might be useful to record any final response details (for example, after changes made by other plugins).
+- [`handlerDidComplete`](/docs/workbox/reference/workbox-core/#method-HandlerDidCompleteCallback): Called after all [extend lifetime promises](https://w3c.github.io/ServiceWorker/#extendableevent-extend-lifetime-promises) added to the event from the invocation of the strategy have settled. This is helpful if you need to report on any data that needs to wait until the handler is done in order to calculate stuff like cache hit status, cache latency, network latency, and other useful information.
+- [`handleDidError`](/docs/workbox/reference/workbox-core/#method-HandlerDidErrorCallback): Called if the handler can't provide a valid response from from _any_ source, which is the optimal time to provide some sort of fallback response as an alternative to failing outright.
+
+All of these callback are `async`, and therefore will be `await`ed whenever a cache or fetch event reaches the relevant point for the callback concerned.
```

Comment:
styleguide suggests not mixing code and non code `await`ed, is there a way to rephrase to avoid this?

Reply:
I fixed this in an inline commit.